File: Sorting/selectionSort.js
// sort the array below using selection sort

const array = [32, 11, 8, 9, 23, 10];

console.log("unsorted array: ", array)

for (let i = 0; i < array.length - 1; i++) {
  let minIndex = i;

  for (let j = i + 1; j < array.length; j++) {
    if (array[j] < array[minIndex]) {
      minIndex = j;
    }
  }
  swap(array, i, minIndex)
}

function swap(arr, firstIndex, secondIndex) {
  const temp = arr[firstIndex];
  arr[firstIndex] = arr[secondIndex];
  arr[secondIndex] = temp;
}

console.log("Sorted Array: ", array)

File: Sorting/src/com/vinayak/SelectionSort.java
package com.vinayak;

import java.util.Arrays;

public class SelectionSort {

    public static void main(String[] args) {
	    // sort the following unsorted array using selection sort

        int[] array = {11, 4, 15, 8, 4, 6, 20};

        System.out.println("unsorted array: " + Arrays.toString(array));

        for (int i = 0; i < array.length - 1; i++) {
            int min = i;
            for (int j = i; j < array.length; j++) {
                if (array[j] < array[min]) {
                    min = j;
                }
            }
            int temp = array[i];
            array[i] = array[min];
            array[min] = temp;
        }
        System.out.println("sorted array: " + Arrays.toString(array));
    }
}
